boot/bootutil/src/loader.c

```diff
@@ -1247,116 +1297,104 @@ boot_go(struct boot_rsp *rsp)
     boot_data.imgs[0].sectors = slot0_sectors;
     boot_data.imgs[1].sectors = slot1_sectors;
 
-    /* Open boot_data image areas for the duration of this call. */
-    for (slot = 0; slot < BOOT_NUM_SLOTS; slot++) {
-        fa_id = flash_area_id_from_image_slot(slot);
-        rc = flash_area_open(fa_id, &BOOT_IMG_AREA(&boot_data, slot));
+    /* The ARMv8-M architecture can contain more than one bootable
+     * image(secure and non-secure). The img_cnt paramter indicates how many
+     * image must be authenticated. Three flash partitions belongs to one
+     * bootable image:
+     *          - primary slot(image always executes from here)
+     *          - secondary slot(partitons for new image during sw upgrade)
+     *          - scratch area(used during image swapping)
+     * The flash_map_entry structure is initilized at the beginning of the
+     * loop to contain the addreses of the corresponding flash partitions,
+     * which belongs the current image.
+     */
+    for (img_idx = 0 ; img_idx < img_cnt; img_idx++) {
+        /* Initialize flash partitoin map */
+        flash_partition_map_init(img_idx);
+
+        /* Open boot_data image areas for the duration of this call. */
+        for (slot = 0; slot < BOOT_NUM_SLOTS; slot++) {
```

Comment:
The language around "images" versus "slots" is getting confused for the reader here; it seems like there are still three "slots" for each of the two "images", but all six of these are "flash areas". This should be cleared up in the design doc.